crates/rpc/src/method/transfer_transaction.rs

```diff
@@ -66,56 +67,52 @@ pub async fn transfer_transaction(
     let mut instructions = vec![];
 
     // Handle native SOL transfers
-    if request.token == NATIVE_SOL {
-        instructions.push(system_instruction::transfer(&source, &destination, request.amount));
-    } else {
-        // Handle wrapped SOL and other SPL tokens
-        validator.validate_token_mint(&token_mint)?;
-
-        let mint_account = rpc_client
-            .get_account(&token_mint)
-            .await
-            .map_err(|e| KoraError::RpcError(e.to_string()))?;
-
-        let mint = Mint::unpack(&mint_account.data)
-            .map_err(|_| KoraError::ValidationError("Invalid mint account data".to_string()))?;
-
-        let decimals = mint.decimals;
-        let source_ata = get_associated_token_address(&source, &token_mint);
-        let dest_ata = get_associated_token_address(&destination, &token_mint);
-
-        let _ = rpc_client
-            .get_account(&source_ata)
-            .await
-            .map_err(|_| KoraError::AccountNotFound(source_ata.to_string()))?;
-
-        if rpc_client.get_account(&dest_ata).await.is_err() {
-            instructions.push(create_associated_token_account(
-                &fee_payer,
-                &destination,
-                &token_mint,
-                &spl_token::id(),
-            ));
+    match token {
+        TokenType::TokenKeg(_) => {
+            instructions.push(system_instruction::transfer(&source, &destination, request.amount));
+        }
```

Comment:
This instruction is for SOL transfers. Please update this to use the token trait to avoid casing on the token types